async/src/connection.rs

```diff
@@ -89,7 +89,7 @@ pub struct Connection {
   /// next request id
   pub request_index:     RequestId,
   /// list of finished requests
-  pub finished_reqs:     HashSet<RequestId>,
+  pub finished_reqs:     HashMap<RequestId, bool>,
```

Comment:
I should add a comment there, but it's there only to mimic what's done for finished_get_reqs so that both can be handled exactly the same way. In the case of finished_reqs it's actually always true, it's just a matter of unifying the two.
I thought of even merging the two, but i left them separated for now.